src/main/java/seedu/duke/command/DeleteFromAvailableCommand.java

```diff
@@ -0,0 +1,84 @@
+package seedu.duke.command;
+
+import seedu.duke.command.DeleteCommand;
+import seedu.duke.data.AvailableModulesList;
+import seedu.duke.data.SelectedModulesList;
+import seedu.duke.data.SemModulesList;
+import seedu.duke.module.SelectedModule;
+import seedu.duke.ui.Ui;
+import seedu.duke.module.Module;
+import seedu.duke.module.NewModule;
+
+public class DeleteFromAvailableCommand extends DeleteCommand {
+
+    private String moduleIdentifier;
+    private String type;
+    /**
+     * Constructor for DeleteFromAvailableCommand.
+     * @param moduleIdentifier The Id or the Name of the Module.
+     * @param type To determine if the moduleIdentifier is an Id or the Name of the module.
+     */
+
+    public DeleteFromAvailableCommand(String moduleIdentifier, String type) {
+        this.moduleIdentifier = moduleIdentifier;
+        this.type = type;
+    }
+
+    public void execute(SelectedModulesList selectedModulesList, AvailableModulesList availableModulesList) {
+        boolean isModuleAvailable = checkIfModuleAvailable(availableModulesList);
+        if (!isModuleAvailable) {
+            Ui.showError(String.format("Module %s not found in available modules", moduleIdentifier));
+            return;
+        }
+
+        Module moduleChosen = availableModulesList.getModule(moduleIdentifier);
+        boolean isPreReq = checkIfIsPreReq(moduleChosen, availableModulesList);
+        if (isPreReq) {
+            Ui.showError(String.format("Module %s cannot be deleted because it is a prerequisite to other modules.",
+                    moduleIdentifier));
+            return;
+        }
+        availableModulesList.remove(moduleChosen); // why did this work????
```

Comment:
Actually, I previously found that the data structure of our `AvailableModulesList` is quite messy. You can simply understand it as there are actually two `availableModulesList` in this class. 

- One is the `AvailableModulesList` self which extends `ModuleList`. 

- But I find there is another `static ModuleList availableModulesList` in the `AvailableModulesList` object. And the constructor add the modules in the database to this `availableModulesList`.

 It is ok to keep the second static list in the object because it is class-level and other methods can get `availableModulesList` easily through this class-level static variable. But it caused a problem we need to notice:

when we use `availableModulesList.add(X)` in other method, actually it is added to the first list(self) I mentioned above. So I override the `add` method in `AvailableModulesList`. 
`    @Override
    public boolean add(Module module) {
        availableModulesList.add(module);
        return super.add(module);
    }`
So to solve the potential problem, you may need to override the delete method in `AvailableModulesList`.